app/src/androidTest/java/com/google/android/perftesting/testrules/MeasureExecutionTime.java

```diff
@@ -5,24 +5,27 @@
 package com.google.android.perftesting.testrules;
 
 import android.os.Trace;
-import android.util.Log;
 
 import org.junit.rules.ExternalResource;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 
 import java.io.BufferedReader;
 import java.io.FileWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import static com.google.android.perftesting.common.PerfTestingUtils.getTestFile;
 
 
-public class GetExecutionTime extends ExternalResource{
+public class MeasureExecutionTime extends ExternalResource {
 
+    private Logger logger = Logger.getLogger(MeasureBatteryStats.class.getName());
     private String mTestName;
     private String mTestClass;
-    private long StartTime;
-    private long EndTime;
+    private long startTime;
+    private Long endTime;
+    private long executionThresholdMs;
```

Comment:
As per our discussion, rename it to `thresholdMillis`. BTW, the prefix `execution` is unnecessary because this rule measures only execution time.